titus-server-master/src/main/java/com/netflix/titus/master/mesos/MesosStateTracker.java

```diff
@@ -40,158 +35,37 @@
  */
 class MesosStateTracker {
 
-    private static final Logger logger = LoggerFactory.getLogger(MesosStateTracker.class);
-
     private static final String TASKS_METRIC_ID = MetricConstants.METRIC_MESOS + "tasks";
-    private static final String TRACKED_TASKS_METRIC_ID = MetricConstants.METRIC_MESOS + "trackedTasks";
 
-    private final MasterConfiguration configuration;
+    private final Cache<String, Protos.TaskStatus> knownTasks;
+    private final Cache<String, Protos.TaskStatus> unknownTasks;
 
-    private final ConcurrentMap<String, TaskTracker> taskTrackers = new ConcurrentHashMap<>();
-    private final Queue<TaskTracker> expiryQueue = new ConcurrentLinkedQueue<>();
-    private final Clock clock;
     private final Registry registry;
 
     MesosStateTracker(MasterConfiguration configuration, TitusRuntime titusRuntime) {
-        this.configuration = configuration;
-        this.clock = titusRuntime.getClock();
         this.registry = titusRuntime.getRegistry();
-        PolledMeter.using(registry).withId(registry.createId(TRACKED_TASKS_METRIC_ID)).monitorSize(taskTrackers);
+
+        this.knownTasks = CacheBuilder.newBuilder()
```

Comment:
FYI, there is a `InstrumentedCache` class that instruments a cache already. You might not care about the other metrics though.